src/gcp_airflow_foundations/source_class/ftp_gcs_source.py

```diff
@@ -0,0 +1,217 @@
+from dataclasses import fields
+from os import X_OK
+from urllib.parse import urlparse
+from abc import ABC, abstractmethod, abstractproperty
+import logging
+import json
+import pandas as pd
+from dacite import from_dict
+from dataclasses import dataclass
+
+from airflow.models.dag import DAG
+from airflow.utils.task_group import TaskGroup
+from airflow.operators.python_operator import PythonOperator
+from airflow.operators.dummy import DummyOperator
+from airflow.providers.google.cloud.transfers.gcs_to_bigquery import GCSToBigQueryOperator
+from airflow.providers.google.cloud.sensors.gcs import GCSObjectExistenceSensor
+from airflow.providers.google.cloud.hooks.gcs import GCSHook
+from airflow.exceptions import AirflowException
+
+from gcp_airflow_foundations.base_class.gcs_table_config import GCSTableConfig
+from gcp_airflow_foundations.base_class.gcs_source_config import GCSSourceConfig
+from gcp_airflow_foundations.operators.api.operators.sf_to_gcs_query_operator import SalesforceToGcsQueryOperator
+from gcp_airflow_foundations.operators.api.sensors.gcs_sensor import GCSObjectListExistenceSensor
+from gcp_airflow_foundations.source_class.ftp_source import FTPtoBQDagBuilder
+from gcp_airflow_foundations.base_class.data_source_table_config import DataSourceTablesConfig
+from gcp_airflow_foundations.source_class.source import DagBuilder
+from gcp_airflow_foundations.common.gcp.load_builder import load_builder
+
+
+class GCSFiletoBQDagBuilder(FTPtoBQDagBuilder):
+    """
+    Builds DAGs to load files from GCS to a BigQuery Table.
+
+    For GCS->BQ ingestion, either a metadata file is required or the field templated_file_name must be provided. 
```

Comment:
right now im only validating that the configs are actually passed - but yes, should be validating some extra logic...left it be for now since we're not entirely sure of the scope yet